src/EGraphs/saturation.jl

```diff
@@ -72,6 +72,12 @@ Base.@kwdef mutable struct SaturationParams
   threaded::Bool                       = false
   timer::Bool                          = true
   printiter::Bool                      = false
+  "Buffer for e-matching which defaults to a global. Use a local buffer for generated functions."
+  buffer::CircularDeque{Bindings}      = BUFFER[]
+  buffer_lock::ReentrantLock           = BUFFER_LOCK
+  "Buffer for rule application which defaults to a global. Use a local buffer for generated functions."
+  merges_buffer::CircularDeque{Tuple{Int,Int}} = MERGES_BUF[]
+  merges_buffer_lock::ReentrantLock    = MERGES_BUF_LOCK
```

Comment:
I think it would be better to pass the buffer not via params, but via e-graph construction. This way we can avoid passing params to the compiled e-matcher functions in `ematch_compiler.jl`

Reply:
Sure I'll move it into the e-graph!